packages/harpguru-core/src/utils/reduce-fullmatrix-to-viewablematrix/reduce-fullmatrix-to-viewablematrix.ts

```diff
@@ -0,0 +1,20 @@
+import type { HarpFaceMatrix } from 'harpparts'
+
+import { isPopulatedArray } from '../is-populated-array'
+import type { ColumnBounds } from '../../types'
+import { sliceMatrix } from '../../packages/slice-matrix'
+
+export const reduceFullMatrixToViewableMatrix = <T>(
+  activeMatrix: HarpFaceMatrix<T>,
+  columnBounds: ColumnBounds
+): HarpFaceMatrix<T> => {
+  if (columnBounds === 'FIT') return activeMatrix
+
+  const [start, end] = columnBounds
+
+  const nextViewableMatrix = sliceMatrix(activeMatrix, start, end + 1).filter(
+    isPopulatedArray
+  )
+
+  return nextViewableMatrix
```

Comment:
This function isn't checking for the equality of the next and previous values like the others are.